migrator/replica/db_replica_manager_impl.go

```diff
@@ -0,0 +1,98 @@
+package replica
+
+import (
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stackrox/rox/migrator/replica/postgres"
+	"github.com/stackrox/rox/migrator/replica/rocksdb"
+	"github.com/stackrox/rox/pkg/features"
+)
+
+// DBReplicaManagerImpl - scans and manage database replicas within central.
+type DBReplicaManagerImpl struct {
+	forceRollbackVersion string
+	adminConfig          *pgxpool.Config
+	sourceMap            map[string]string
+	basePath             string
+	dbmRocks             *rocksdb.DBReplicaManagerImpl
+	dbmPostgres          *postgres.DBReplicaManagerImpl
+}
+
+// NewPostgres - returns a new ready-to-use store.
+func NewPostgres(basePath string, forceVersion string, adminConfig *pgxpool.Config, sourceMap map[string]string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		forceRollbackVersion: forceVersion,
+		adminConfig:          adminConfig,
+		sourceMap:            sourceMap,
+		basePath:             basePath,
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+		dbmPostgres:          postgres.New(forceVersion, adminConfig, sourceMap),
+	}
+}
+
+// New - returns a new ready-to-use store.
+func New(basePath string, forceVersion string) *DBReplicaManagerImpl {
+	return &DBReplicaManagerImpl{
+		forceRollbackVersion: forceVersion,
+		basePath:             basePath,
+		dbmRocks:             rocksdb.New(basePath, forceVersion),
+	}
+}
+
+// Scan - checks the persistent data of central and gather the replica information
+// from disk.
+func (d *DBReplicaManagerImpl) Scan() error {
+	err := d.dbmRocks.Scan()
+	if err != nil {
+		return err
+	}
+
+	if features.PostgresDatastore.Enabled() {
+		err = d.dbmPostgres.Scan()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// GetReplicaToMigrate - finds a replica to migrate.
+// It returns the replica link, path to database, postgres database name and error if fails.
+func (d *DBReplicaManagerImpl) GetReplicaToMigrate() (string, string, string, error) {
+	log.Info("GetReplicaToMigrate")
+	// Get the RocksDB replica we are migrating
+	replica, replicaPath, err := d.dbmRocks.GetReplicaToMigrate()
+	if err != nil {
+		return "", "", "", err
+	}
+
+	if features.PostgresDatastore.Enabled() {
+		// Get the version of the Rocks replica so Postgres manager can use that info
+		// to determine what replica it needs to migrate.
+		rocksVersion := d.dbmRocks.GetVersion(replica)
+
+		pgReplica, migrateFromRocks, err := d.dbmPostgres.GetReplicaToMigrate(rocksVersion)
+		if err != nil {
+			return "", "", "", err
+		}
+
+		// If we need to migrate from rocks we need to return both
+		// rocks and postgres replicas.  Otherwise we just need one or
+		// the other.
+		if migrateFromRocks {
+			return replica, replicaPath, pgReplica, err
+		}
+		return "", "", pgReplica, err
+
+	}
+
+	return replica, replicaPath, "", err
+}
+
+// Persist - replaces current replica with upgraded one.
+func (d *DBReplicaManagerImpl) Persist(replicaName string) error {
```

Comment:
Based on existing logic, if rollback is enabled, we are creating a TempReplica from Current.  This TempReplica is the replica we are working with through the migration process.  When we go to persist AND are in Postgres we do this
`if persistBoth && replicaName == rocksdb.TempReplica {
	if err := d.dbmRocks.Persist(replicaName); err != nil {
		log.Warnf("Unable to create a previous version of Rocks to rollback to: %v", err)
	}
}`

That will take the original current and move it to previous.  It will also move temp to the new current on Rocks.  What we wind up with is a previous with a pre-migration version and a current that has the updated migration sequence number.  I think the previous logic is correct.  I will run some tests on the current part as that may not be completely correct.  There are checks elsewhere that require a current Rocks to exist I was trying to avoid messing with those as there is a huge ripple if I do.

Reply:
@c-du After looking at this case some more, it is tough since i have migrations turned off to see the permutations of how the version works its way through the versions.  So I would like to either defer this case until ROX-11922 OR if I'm still writing unit tests when your change is merged, I can try to incorporate that in this PR.  I have a good idea of how I'll make it work so it shouldn't be that bad.

Reply:
I have turned migrations on and I believe I have covered this scenario